lib/pbench/client/oidc_admin.py

```diff
@@ -0,0 +1,150 @@
+from http import HTTPStatus
+import json
+import os
+
+import requests
+
+from pbench.server.auth import Connection
+
+
+class OIDCAdmin(Connection):
```

Comment:
The way this is implemented will certainly work, but it's not really "correct":  I don't see any indication that an instance of the `OIDCAdmin` class _is an_ instance of the `Connection` class.  That is, the code never accesses any inherited methods or attributes, and the `OIDCAdmin` class doesn't override any methods from the `Connection` class.

So, the proper implementation is for an instance of the `OIDCAdmin` class to _contain_ an instance of the `Connection` class.  That is, at line 16, instead of calling `super().__init__()`, it should just create an instance of the `Connection` class and store it in an attribute of the `OIDCAdmin` class.  Then, when the methods of the `OIDCAdmin` class want to invoke methods of the `Connection` class, they do so on the attribute instead of on `self`.  (It means slightly more characters on the screen, but it's just as efficient in execution, and it's a more-pure/proper relationship between the classes.)

Reply:
> The way this is implemented will certainly work, but it's not really "correct": I don't see any indication that an instance of the OIDCAdmin class is an instance of the Connection class. That is, the code never accesses any inherited methods or attributes, and the OIDCAdmin class doesn't override any methods from the Connection class.

I wouldn't consider that a fully accurate assessment of "correctness". To me it seems that this is a proper extension of `Connection`, which adds additional specialized methods on top of the base HTTP methods and connection tracking. I don't see any advantage in hiding that infrastructure here. That is, being able to `post` to the OIDC server through an `OIDCAdmin` in addition to using the specialized extensions makes logical sense and it would be pointless to invent new wrappers to pass that through.

Reply:
You are correct:  we cannot really assess "correctness" without requirements, and I was trying to infer the requirements based on the implementation.  Instead, we really should ask, "Do we need/want to be able to use an instance of an `OIDCAdmin` as though it _were_ a `Connection`, or is the fact that it uses a `Connection` simply a(n internal) detail of the `OIDCAdmin` implementation?"  Obviously, I believe that the latter is the case, that we have specific methods implemented on `OIDCAdmin` which are implemented via method calls on a `Connection` instance, and that the `OIDCAdmin` instance is its own, independent thing (and therefore that it should _contain_ and not be derived from `Connection`).

In any case, I don't intend to gate the merge on this point.